### ISSUE TITLE
Accessibility: the use of autocomplete=”new-password” is a problem

### ISSUE THREAD

**Issue by Zodiac1978** [pluginkollektiv/antispam-bee#648]
```
> Actually, the autocomplete attribute allow some disabled users to use tools that will autocomplete fields with their data saved in their browser. It can also display related icons next to the fields so that people with cognitive or intellectual disabilities can fill the field easily.

On [the link I gave](https://www.w3.org/TR/WCAG22/#input-purposes), this is written here (before the list):

> The terms below are not keywords that must be used, but instead represent purposes that must be captured in the taxonomy adopted by a webpage. Where applicable, authors mark up controls with the chosen taxonomy to indicate the semantic purpose. This provides the potential for user agents and assistive technologies to apply personalized presentations that can enable more people to understand and use the content.

And you also can read the example I gave in [the WCAG criterion “Identify input purpose”](https://www.w3.org/WAI/WCAG22/Understanding/identify-input-purpose.html) (if you search for “icon”, you will have more information in the page):

> By adopting and repurposing this predefined taxonomy of definitions, user agents and assistive technologies can now present the purpose of the inputs to users in different modalities. For example, assistive technologies may display familiar icons next to input fields to help users who have difficulties reading. An icon of a birthday cake may be shown in front of an input field with autocomplete="bday", or the icon of a telephone in front of an input field with autocomplete="tel".

So, if you have a comment field with an icon representing password, this is confusing for people.
```

From: https://wordpress.org/support/topic/accessibility-the-use-of-autocompletenew-password-is-a-problem/

**Comment by 00travelgirl00** [comment]
@Zodiac1978 thank you for tagging me and raising awareness about this problem. Unfortunately, I don't have experience with autocomplete and browser support, so I can't really help you out.

Is there another way to turn off autocomplete that all browsers will respect?

**Comment by addedlovely** [comment]
Hey, this just got flagged in our accessibility audit.  Not sure why you set this, but would be preferable to just not set autocomplete at all on the comment texarea /  message. 

Or you can try:
`autocomplete="off"`

What were you trying to work around by having a nonstandard value on a field that doesn't need it?

**Comment by Zodiac1978** [comment]
> What were you trying to work around by having a nonstandard value on a field that doesn't need it?

This has a backstory with a Chrome bug, which fills input fields with wrong values. We needed to find a way to stop this and as "autocomplete=off" was ignored by Chrome we ended up with "new-password", which is _not_ nonstandard.

See:
https://github.com/pluginkollektiv/antispam-bee/issues/280
https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/autocomplete#new-password

**Comment by addedlovely** [comment]
@Zodiac1978 Sorry, should have need clearer - can see it set on the honeypot trap ( which I assume our audit has ignored as it's that's marked aria-hidden and is a non issue ) - but it's leaking onto the actual comment field text area too.  The markup below is from a WooCommerce product review form.

I assume that's not intentional?

```
<textarea autocomplete="new-password"  id="jc146aee42"  name="jc146aee42"   cols="45" rows="8" required placeholder="Your review"></textarea>
<textarea id="comment" aria-label="hp-comment" aria-hidden="true" name="comment" autocomplete="new-password" style="padding:0 !important;clip:rect(1px, 1px, 1px, 1px) !important;position:absolute !important;white-space:nowrap !important;height:1px !important;width:1px !important;overflow:hidden !important;" tabindex="-1"></textarea>
```

**Comment by Zodiac1978** [comment]
Yes, this is intentional.It was bug in Chrome 9 years ago. Street was inserted as comment:
https://wordpress.org/support/topic/merkwurdiges-verhalten/

You can try to remove the attribute from this line:
https://github.com/pluginkollektiv/antispam-bee/blob/54cb2c155521c91020087d99cd8000d6a4af465a/antispam_bee.php#L1287

In my test case this does not lead to the bug anymore. Maybe we can safely remove this workaround after 9 years.

**Comment by addedlovely** [comment]
Thanks, was expecting the Chromium bug report to be resolved, but it's a nightmare - even worse than an old WordPress trac ticket.

Chrome doesn't look to still exhibit that behaviour, at least in English. 
For anyone else looking, can just hook in and remove and add autocomplete="none" as required.
( Reads like Chrome ignores 'none' regardless, but for other more sane browsers should work ok )

```
/**
 * Comments : Force autocomplete="off" on comment textarea field.
 * Added by AntiSpam Bee as a Chrome workaround, fails accessibility audit.
 */
function addedlovely_comment_form_field_comment( $comment_field ) {
	// Replace any existing autocomplete value on the textarea with "off"
	$comment_field = preg_replace(
		'/(<textarea\b[^>]*)\sautocomplete=["\'][^"\']*["\']/i',
		'$1 autocomplete="off"',
		$comment_field,
		1
	);

	// Safety net: inject if no autocomplete attribute is present at all
	if ( ! preg_match( '/autocomplete=/i', $comment_field ) ) {
		$comment_field = preg_replace( '/<textarea\b/i', '<textarea autocomplete="off"', $comment_field, 1 );
	}

	return $comment_field;
}
add_filter( 'comment_form_field_comment', 'addedlovely_comment_form_field_comment', 100, 1 );

```
